src/classes/ALLO_PaymentSync_TEST.cls

```diff
@@ -221,14 +221,143 @@ private with sharing  class ALLO_PaymentSync_TEST {
         alloForInsert[0].Amount__c = 250;
     }
 
-//TODO written off payments aren't counted in the sync
-//TODO Paid payment allocations aren't updated with opportunity allocations changes
-    //TODO can't lower or delete an opp allocation for a GAU that has paid payment allocations
-//TODO updating paid payment allocations does update opportunity allocations
-//TODO null and zero amounts for both opps and payments
-//TODO negative payment amounts
+    /*******************************************************************************************************
+    * @description When changing the amount of a Written off Payment with Allocations: 
+    * Changes to the payment values will not change the opportunity allocations
+    ********************************************************************************************************/
+    @isTest
+    static void doNotSyncWrittenOffPaymentToOpportunity() {
+        makePayments(1000, 1);
+        for(npe01__OppPayment__c payment: payments) {
+            payment.npe01__Written_Off__c = true;
+        }
+        update payments;
+        makeGAUs(2);
+
+        System.assert(UTIL_CustomSettingsFacade.getAllocationsSettings().Payment_Allocations_Enabled__c, 
+            UTIL_CustomSettingsFacade.getAllocationsSettings());
+
+        //resetting these manually as a test is considered a single context for static variables
+        List<Allocation__c> alloForInsert = new List<Allocation__c>();
+
+        alloForInsert.add(new Allocation__c(Payment__c=payments[0].Id, Percent__c = 50, General_Accounting_Unit__c = gaus[0].Id));
+        alloForInsert.add(new Allocation__c(Payment__c=payments[0].Id, Amount__c = 300, General_Accounting_Unit__c = gaus[1].Id));
+        insert alloForInsert;
+
+        List<Allocation__c> queryAllocations = getAllocationsOrderByAmount();
+        System.assertEquals(3, queryAllocations.size(), 'Three allocations should exist.');
+
+        // no allocations should have been created for opportunity
+        System.assertEquals(null, queryAllocations[0].Opportunity__c, 'Only Payment allocations should have been created');
+        System.assertEquals(null, queryAllocations[1].Opportunity__c, 'Only Payment allocations should have been created');
+        System.assertEquals(null, queryAllocations[2].Opportunity__c, 'Only Payment allocations should have been created');
+
+        //original payment allocations
+        assertAllocation(queryAllocations[0], 200, defaultGau.Id, false);
+        assertAllocation(queryAllocations[1], 300, gaus[1].Id, false);
+        assertAllocation(queryAllocations[2], 500, gaus[0].Id, false);
+    }
+
+
+    /*******************************************************************************************************
+    * @description When changing the amount of Opportunity allocations connected to paid payments
+    * Payment Allocations for paid payments will not be changed
```

Comment:
It is good either way, but maybe could you add one additional payment that is not paid, and confirm that the changes do flow to that payment?

Reply:
It's a good point - I could definitely be flexible, but I'm trying to keep my unit tests more focused and atomic for what they are testing in general (and in this case stick to the specific tests that Nic was requesting). I might consider that as a separate test method if anything, more than putting that test in here.

Reply:
@ptewson-sfdo - Maybe we log these as a future work item, and a good candidate for a new developer to cut their teeth on.